notesapi/v1/models.py

```diff
@@ -0,0 +1,141 @@
+
+        self.ranges = json.dumps(ranges)
+
+    def as_dict(self):
+        """
+        Returns the note object as a dictionary.
+        """
+        created = self.created.isoformat() if self.created else None
+        updated = self.updated.isoformat() if self.updated else None
+
+        return {
+            'id': self.pk,
+            'user': self.user_id,
+            'course_id': self.course_id,
+            'usage_id': self.usage_id,
+            'text': self.text,
+            'quote': self.quote,
+            'ranges': json.loads(self.ranges),
+            'created': created,
+            'updated': updated,
+        }
+
+
+@receiver(signals.post_save, sender=Note)
+def update_in_index(sender, instance, **kwargs):
+    if settings.ES_DISABLED:
+        return
+    NoteMappingType.index(instance.as_dict(), id_=instance.id, overwrite_existing=True)
+
+
+@receiver(signals.post_delete, sender=Note)
+def delete_in_index(sender, instance, **kwargs):
+    if settings.ES_DISABLED:
+        return
+    NoteMappingType.unindex(id_=instance.id)
+
+
+class NoteMappingType(MappingType, Indexable):
+    """
+    Mapping type for Note.
+    """
+
+    @classmethod
+    def get_model(cls):
+        return Note
+
+    @classmethod
+    def get_mapping(cls):
+        """
+        Returns an Elasticsearch mapping for Note MappingType
+        """
+        charfield = {'type': 'string', 'index': 'not_analyzed', 'store': True}
+        return {
+            'properties': {
+                'id': charfield,
+                'user': charfield,
+                'course_id': charfield,
+                'usage_id': charfield,
+                'text': {'type': 'string', 'analyzer': 'snowball', 'store': True},
+                'quote': {'type': 'string', 'analyzer': 'snowball', 'store': True},
+                'created': {'type': 'date', 'store': True},
+                'updated': {'type': 'date', 'store': True},
+            }
+        }
+
+    @classmethod
+    def extract_document(cls, obj_id, obj=None):
+        """
+        Converts this instance into an Elasticsearch document.
+        """
+        if obj is None:
+            obj = cls.get_model().objects.get(pk=obj_id)
+
+        return obj.as_dict()
+
+    @staticmethod
+    def process_result(data):
+        """
+        Prepares result for response.
+
+        Also prepares ranges field and highlighting.
+        """
+        for i, item in enumerate(data):
+            if isinstance(item, dict):
+                for k, v in item.items():
+                    if k != 'ranges' and isinstance(v, list) and len(v) > 0:
+                        data[i][k] = v[0]
+
+                    # Substitute the value of text field by highlighted result.
+                    if len(item.es_meta.highlight) and k == 'text':
+                        data[i][k] = item.es_meta.highlight['text'][0]
+
+        return data
+
+note_searcher = NoteMappingType.search()
```

Comment:
That is an object which will be searching through NoteMappingType documents, documented at http://elasticutils.readthedocs.org/en/latest/django_api.html?highlight=search#elasticutils.contrib.django.MappingType.search 

similar to django's SomeModel.objects